### ISSUE TITLE
How can I concatenate Width-Wise in NVDLA since RUBIK doesn't support that?

### ISSUE THREAD

**Issue by number161** [nvdla/sw#144]
I am trying to implement YoloV3 in NVDLA. There is a concatenation operation in YOLOv3. I cannot find any module in nvdla that supports this.
Rubik supports only channel-wise splitting and merging. I need to concatenate data cubes width-wise.

**Comment by prasshantg** [comment]
No HW operation required for this. This can be achieved using single tensor and setting correct strides. NVDLA compiler handles it.

**Comment by xiao777dong** [comment]
> No HW operation required for this. This can be achieved using single tensor and setting correct strides. NVDLA compiler handles it.

Hi,  prasshantg. How should I do to concate two parts? How to set correct strides? Hoping for help. Thank you.

**Comment by xiao777dong** [comment]
> I am trying to implement YoloV3 in NVDLA. There is a concatenation operation in YOLOv3. I cannot find any module in nvdla that supports this.
> Rubik supports only channel-wise splitting and merging. I need to concatenate data cubes width-wise.

I have met the same problem. Have you solved the problem? Thank you.

**Comment by number161** [comment]
I think I have found a solution. You can see this on paper for better visualisation, in concatenation simply the new block is added behind the given block. If you see how this given block is mapped in memory you will see the solution.
Try to visualize how the final concatenated block will be mapped in memory, then you will see that concatenation simple means mapping the two memory blocks back to back in the memory and then when you perform any operation on that block, it will be like you are performing the operation on the final concatenated product(Block). 
So basically I'm viewing this concatenation problem like a memory mapping problem.
I'm afraid addition hardware might be required since we won't be able to manipulate the instruction set of memory just for our "special case". 

![concat](https://user-images.githubusercontent.com/29134859/63246037-58534c80-c27f-11e9-92e3-2adf75bfee84.jpg)

**Comment by xiao777dong** [comment]
Thank you very much.
